### ISSUE TITLE
Advanced Rocketry refrences biomesoplenty 

### ISSUE THREAD

**Issue by officereso** [Nomi-CEu/Nomi-CEu#1156]
### Nomi CEu Version

1.7.3

### Launcher

Prism Launcher

### Environment

Singleplayer

### Mode

Both Modes

### What Happened

Venus and Mercury both use overworld biomes for generation. 

### Expected Behavior

Venus and Mercury should use Advanced Rocketry biomes or include BiomesOPlenty for using their biomes.

### Reproduction Steps

1. Download the pack from curseforge
2. Create a new singleplayer world with cheats
3. run ``/advRocketry goto 102`` to goto Venus

### Logs

[2024-12-16-1.log](https://github.com/user-attachments/files/18157013/2024-12-16-1.log)
On line 5246 of this log file you can see advanced rocketry fail to find the biomesoplenty biome.

### Additional Information

This is likely caused by: 
The culprit lines are here in the config 
https://github.com/Nomi-CEu/Nomi-CEu/blob/13a70172da4e2dac6531b505390280da905c824c/overrides/config/advRocketry/planetDefs.xml#L36
https://github.com/Nomi-CEu/Nomi-CEu/blob/13a70172da4e2dac6531b505390280da905c824c/overrides/config/advRocketry/planetDefs.xml#L26

The config also references ``nuclearcraft:fluid.sulfuric_acid`` and ``minecraft:fluid.lava`` neither of which are valid.

I believe when advRocketry fails to populate any biomes from planetDefs.xml it ends up using the overworld biomes instead.

**Comment by IntegerLimit** [comment]
They way I see it, we really have two options:
1. Remove those lines, and figure out if that means we must stick with overworld, or if we can move into Adv Rocketry biomes
2. Add Biomes O Plenty to Nomi-CEu

I'm not sure which at the moment, I see merits with both solutions. As this is a worldgen issue, moving to 1.8.

**Comment by officereso** [comment]
In a fork I had added Biomes O Plenty with no overworld generation to solve this issue. I also used Just Enough Dimensions to create a Biomes O Plenty planet and added that ta Adv Rocketry.

Just throwing this in as a possible idea.

**Comment by officereso** [comment]
I started having an incompatibility issue with BoP and Nomi Labs. 

Attempting to place any of the BoP plants causes a crash.

It seems Nomi Labs injects some code into ftbutils `onRightClickBlock` function which ends up calling the `isReplaceable` function on the held BoP plant. 
That call is made here:
 https://github.com/Nomi-CEu/Nomi-Labs/blob/1d687175c093c02aa95354da725b4d2857f36c42/src/main/java/com/nomiceu/nomilabs/mixin/ftbutilities/FTBUtilitiesPlayerEventHandlerMixin.java#L53
The `isReplaceable` function for the BoP plant casts the return of `world.getBlockState(rightClickedBlock).getValue(heldItem)` to `BOPPlants`. I'm paraphrasing the code here. 
Problem is if you are clicking on something that isnt a BOPPlant this errors and crashes.

Here's the specific BoP function that's problematic:
 https://github.com/Glitchfiend/BiomesOPlenty/blob/5ad77116382bec782a023926b8dab17a6ea5599d/src/main/java/biomesoplenty/common/block/BlockBOPPlant.java#L287

From what I can tell this is a BoP bug and would need to be addressed if BoP was added. I'm very doubtful they would update the 1.12.x version of the mod to fix it.

This should be fixable via MixinBooter though.

**Comment by IntegerLimit** [comment]
Yep, thank you for the notice! I'll fix this via mixin for the newest Labs release.

**Comment by IntegerLimit** [comment]
Biomes O Plenty crash fixed in Labs Update in [this commit](https://github.com/Nomi-CEu/Nomi-CEu/commit/678c9143734e96ae234bd4aa36f72f79a3906a52).

**Comment by officereso** [comment]
Ive been trying to polish up my fork which has BoP in it, Ill draft a pr here has a potential solution for closing this issue when and if my fork is good enough. 
Unfortunately BoP generates its own ores which seem to be really troublesome to remove. As far as I can tell any changes to the default biome config files simply get reset on init. [See here](https://github.com/Glitchfiend/BiomesOPlenty/blob/5ad77116382bec782a023926b8dab17a6ea5599d/src/main/java/biomesoplenty/common/init/ModBiomes.java#L205) and [here](https://github.com/Glitchfiend/BiomesOPlenty/blob/5ad77116382bec782a023926b8dab17a6ea5599d/src/main/java/biomesoplenty/common/init/ModBiomes.java#L263). 
It seems to me that the BoP and Vanilla biomes are hard coded into the mod and will be used no matter what config changes are made. I could be wrong here.
The easiest way to disable ore gen for BoP is to hard code the BoP GeneratorOreBase to generate 0 ore per chunk via mixin. 
https://github.com/Glitchfiend/BiomesOPlenty/blob/5ad77116382bec782a023926b8dab17a6ea5599d/src/main/java/biomesoplenty/common/world/generator/GeneratorOreBase.java#L46
The better solution would be to rewrite and un hardcode the BoP biomes _(probably pretty time consuming, I don't think BoP uses any config files for creating biomes)_. Or at least remove the code that adds the each ore generator to the biomes such as [here](https://github.com/Glitchfiend/BiomesOPlenty/blob/5ad77116382bec782a023926b8dab17a6ea5599d/src/main/java/biomesoplenty/common/biome/overworld/BiomeGenFen.java#L111)

At the point of putting this much work into getting BoP into the pack to solve this issue, it is certainly easier to just switch the two advRocketry references to be advRocketry biomes similar to the other planets. It's also possible that #1124 will end up changing the advRocketry biomes and specific planet world generation all together. 
Changing to BoP biomes for 2 planets for them to possibly change later on doesn't make a whole lot of sense to me unless part of polishing the advRocketry part of Nomi is dependent of having specific BoP biomes for planets.

**Comment by IntegerLimit** [comment]
I don't mind putting some work into BoP to add more diversity to planets, that could make exploring actually interesting.

We'll see when we approach AR discussions in 1.8. Thank you for this information!
